CloudFormationTemplate.yml

```diff
@@ -0,0 +1,205 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: >
+  This CloudFormation template creates simple Lambda functions,
+  which prints CloudFormation resource Arn from the stack.  
+Resources:
+  LambdaFunctionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service:
+              - lambda.amazonaws.com
+          Action:
+            - sts:AssumeRole
+      Path: "/"
+      Policies:
+      - PolicyName: LambdaFunctionPolicy
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+          - Effect: Allow
+            Action:
+              - logs:CreateLogGroup
+              - logs:CreateLogStream
+              - logs:PutLogEvents
+              - dynamodb:scan
+              - dynamodb:list
+              - s3:ListAllMyBucket
+              - s3:ListBucket
+              - s3:PutObject
+              - s3:PutObjectAcl
+              - s3:GetBucketLocation
+            Resource: 
+            - arn:aws:dynamodb:eu-west-2:251045990954:table/MetadataJson
+            - arn:aws:lambda:eu-west-2:251045990954:function:Stack-10-LambdaFunction-HUcUFXbCq4ZJ
+            - arn:aws:s3:::dbmetadata/
+            - '*'
+  LambdaFunction:
+    Type: AWS::Lambda::Function
+    Properties:
+      Runtime: python3.7
+      Timeout: 5
+      Handler: index.lambda_handler
+      Role: !GetAtt LambdaFunctionRole.Arn
+      Code:
+        ZipFile:
+          !Sub
```

Comment:
no need of this !Sub
you can just have 
```yaml
Code:
     ZipFile: |
           def lambda_handler(event, context):
                 ## Code
                 return event
``` 

But I would advice against having code within the same template. Please split the code to a different file `index.py`
and reference the lambda code by either keeping the zip in s3. Reference doc - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-function-code.html#aws-properties-lambda-function-code--examples


Or you can make use of aws cloudformation package api 
https://catalog.workshops.aws/cfn101/en-US/intermediate/templates/package-and-deploy
https://docs.aws.amazon.com/cli/latest/reference/cloudformation/package.html